keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -209,6 +215,14 @@ class ResourceActuator(
     }
   }
 
+  private fun DeliveryConfig.isPromotionCheckStale(): Boolean {
+    val age = Duration.between(
+      deliveryConfigRepository.deliveryConfigLastChecked(this),
+      clock.instant()
+    )
+    return age > Duration.ofMinutes(5)
+  }
```

Comment:
Seems a reasonable threshold to me 🤷

Reply:
This means a five-minute latency between un-pausing an app and seeing actuation start, right? That feels a little long-ish to me. (Maybe two minutes instead?) 

The problem is, I don't have a sense of the worst-case delay for the environment check to happen, so I don't know what the threshold is for too small a waiting time. At some point, we should do some benchmarking to get a sense of how long it typically takes keel to do a full sweep of environment checks.

Reply:
I should probably make this configurable but, it's not a 5 minute delay. If the environment check is older than 5 minutes, we just put the resource check to the back of the line. The theory being the environment check is likely happening _right now_ or is about to. Next time the resource is checked the environment check will likely be up to date. That should be a lot sooner than 5 minutes.